File: dart_sample/lib/demos/cupertino/cupertino_text_field_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoTextFieldDemoPage extends StatefulWidget {
  const CupertinoTextFieldDemoPage({super.key});

  @override
  State<CupertinoTextFieldDemoPage> createState() =>
      _CupertinoTextFieldDemoPageState();
}

class _CupertinoTextFieldDemoPageState
    extends State<CupertinoTextFieldDemoPage> {
  final TextEditingController _controller = TextEditingController();
  String _value = '';

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Cupertino text fields',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Rounded, borderless, form validation, multiline, disabled, '
          'attachment, and clear-button states.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        CupertinoTextField(
          controller: _controller,
          placeholder: 'Search settings',
          prefix: const Padding(
            padding: EdgeInsets.symmetric(horizontal: 6),
            child: Icon(CupertinoIcons.search, size: 18),
          ),
          clearButtonMode: OverlayVisibilityMode.editing,
          onChanged: (String value) => setState(() => _value = value),
        ),
        Text(
          _value.isEmpty ? 'No query' : 'Query: $_value',
          style: const TextStyle(fontSize: 12, color: Color(0xFF607D8B)),
        ),
        const CupertinoTextField.borderless(
          placeholder: 'Borderless multiline note',
          minLines: 2,
          maxLines: 4,
          decoration: BoxDecoration(
            color: Color(0xFFF2F2F7),
            borderRadius: BorderRadius.all(Radius.circular(8)),
          ),
        ),
        Form(
          child: CupertinoTextFormFieldRow(
            prefix: const Text('Email'),
            placeholder: 'name@example.com',
            keyboardType: TextInputType.emailAddress,
            autovalidateMode: AutovalidateMode.onUserInteraction,
            validator: (String? value) =>
                value?.contains('@') ?? false ? null : 'Enter a valid email',
          ),
        ),
        const CupertinoTextField(enabled: false, placeholder: 'Disabled field'),
      ],
    );
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }
}

File: src/Sample/Plumix.Sample/Demos/Cupertino/CupertinoTextFieldDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/cupertino_text_field_demo_page.dart

public sealed class CupertinoTextFieldDemoPage : StatefulWidget
{
    public override State CreateState() => new CupertinoTextFieldDemoPageState();
}

internal sealed class CupertinoTextFieldDemoPageState : State
{
    private readonly TextEditingController _controller = new();
    private string _value = string.Empty;

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12.0,
            children:
            [
                new Text("Cupertino text fields", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Rounded, borderless, multiline, disabled, attachment, placeholder, and clear-button states.",
                    fontSize: 14.0,
                    color: Color.FromUInt32(0x8A000000)),
                new CupertinoTextField(
                    controller: _controller,
                    placeholder: "Search settings",
                    prefix: new Padding(
                        EdgeInsetsGeometry.Symmetric(horizontal: 6.0),
                        new Icon(CupertinoIcons.Search, size: 18.0)),
                    clearButtonMode: OverlayVisibilityMode.Editing,
                    onChanged: value => SetState(() => _value = value)),
                new Text(
                    string.IsNullOrEmpty(_value) ? "No query" : $"Query: {_value}",
                    fontSize: 12.0,
                    color: Color.FromUInt32(0xFF607D8B)),
                CupertinoTextField.Borderless(
                    placeholder: "Borderless multiline note",
                    minLines: 2,
                    maxLines: 4,
                    decoration: new BoxDecoration(
                        Color: Color.FromUInt32(0xFFF2F2F7),
                        BorderRadius: BorderRadius.Circular(8.0))),
                new CupertinoTextField(
                    enabled: false,
                    placeholder: "Disabled field"),
            ]);
    }

    public override void Dispose()
    {
        _controller.Dispose();
        base.Dispose();
    }
}
